skimage/draw/draw.py

```diff
@@ -764,3 +764,76 @@ def rectangle(start, end=None, extent=None, shape=None):
     coords = np.meshgrid(*[np.arange(st, en) for st, en in zip(tuple(tl),
                                                                tuple(br))])
     return coords
+
+
+def rectangle_perimeter(start, end=None, extent=None, shape=None, clip=False):
+    """Generate coordinates of pixels that are exactly around a rectangle.
+
+    Parameters
+    ----------
+    start : tuple
+        Origin point of the inner rectangle, e.g., ``(row, column)``.
+    end : tuple
+        End point of the inner rectangle ``(row, column)``.
+        Either `end` or `extent` must be specified.
+    extent : tuple
+        The extent (size) of the inner rectangle.  E.g.,
+        ``(num_rows, num_cols)``.
+        Either `end` or `extent` must be specified.
+    shape : tuple, optional
+        Image shape used to determine the maximum bounds of the output
+        coordinates. This is useful for clipping perimeters that exceed
+        the image size. By default, no clipping is done.
+    clip : bool, optional
+        Whether to clip the perimeter to the provided shape. If this is set
+        to True, the drawn figure will always be a closed polygon with all
+        edges visible.
+
+    Returns
+    -------
+    coords : array of int, shape (Ndim, Npoints)
+        The coordinates of all pixels in the rectangle.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage.draw import rectangle_perimeter
+    >>> img = np.zeros((5, 6), dtype=np.uint8)
+    >>> start = (2, 3)
+    >>> end = (3, 4)
+    >>> rr, cc = rectangle_perimeter(start, end=end, shape=img.shape)
+    >>> img[rr, cc] = 1
+    >>> img
+    array([[0, 0, 0, 0, 0, 0],
+           [0, 0, 1, 1, 1, 1],
+           [0, 0, 1, 0, 0, 1],
+           [0, 0, 1, 0, 0, 1],
+           [0, 0, 1, 1, 1, 1]], dtype=uint8)
+
+    >>> img = np.zeros((5, 5), dtype=np.uint8)
+    >>> rr, cc = rectangle_perimeter(start, end=(10, 10), shape=img.shape,
+                                     clip=True)
+    >>> img[rr, cc] = 1
+    >>> img
+    array([[0, 0, 0, 0, 0],
+           [0, 0, 1, 1, 1],
+           [0, 0, 1, 0, 1],
+           [0, 0, 1, 0, 1],
+           [0, 0, 1, 1, 1]], dtype=uint8)
+
+    """
+
+    if extent is not None:
+        end = np.array(start) + np.array(extent)
+    elif end is None:
+        raise ValueError("Either `end` or `extent` must be given")
+    top_left = np.minimum(start, end)
+    bottom_right = np.maximum(start, end)
+    if extent is None:
+        bottom_right += 1
+    top_left -= 1
+    r = [top_left[0], top_left[0], bottom_right[0], bottom_right[0],
```

Comment:
This is not a valid perimiter for a 3D+ box, is it? I mean `Ndim` in the docstring doesn't actually represent arbitrary number of dimensions, as we only support 2D in this implementation.

Reply:
Yes. I did not worry about 3D boxes. Should I? All the other functions except for `rectangle` support only 2D.

Reply:
Fixed docstring to specify only 2 dimensions.

Reply:
>I did not worry about 3D boxes. Should I?

I wouldn't :). 2D is arguably the most common case. However, I'm pretty sure that some users will find 3D box perimeter drawing function useful (@jni ?).
I suppose it shouldn't be too complex implementing this (we have only to decide on the order of iteration for edges).

Reply:
Personally, my personal feel is that I shouldn't implement the 3D boxes if I'm not going to put the effort in thinking about it (which I have little motivation for at the moment).

I'll leave it to the next person to take on the torch.